libs/core/src/lib/info-label/info-label.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input, OnInit, ElementRef, OnChanges } from '@angular/core';
+import { CssClassBuilder, applyCssClass } from '../utils/public_api';
+
+type LabelType = 'numeric' | 'only-icon' | 'icon';
+
+@Component({
+    // tslint:disable-next-line:component-selector
+    selector: '[fd-info-label]',
+    template: `<ng-content></ng-content>`,
+    styleUrls: ['./info-label.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InfoLabelComponent implements OnInit, OnChanges, CssClassBuilder {
+  
+    /** 
+     * The LabelType represented by the info label .
+     * Can be one of the following: 'numeric' | 'only-icon' | 'icon'
+     * For default info label omit this property
+     */
+    _labelType: LabelType;
+    
+    @Input()
+    set labelType(value: LabelType) {
+        this._labelType = value;
+        this.buildComponentCssClass();
+    }
+
+    get labelType(): LabelType {
+        return this._labelType;
+    }
+
+    /** define the icon type */
+    _glyph: string; 
+
+    @Input()
+    set glyph(value: string) {
```

Comment:
We shoudl be more clear when to use setters and getters and in this case it does not make much sense here. Usually you use setters (or ngOnChange) for specific properly when it can over component lifecycle. After it is initialized. I dont think any of these properties can change after its created. 

Instead of having 1 call at the end once view is initialized, you make here 5 different calls.

Reply:
https://fundamental-ngx.netlify.com/#/core/button

Our Playground is an example where you can see `glyph` property being changed. Another case would be the RTL support where you might want to change the icon ( `arrow-left` to `arrow-right`)